nssp_demo/collate_plots.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+
+import argparse
+import fnmatch
+import logging
+import os
+from pathlib import Path
+
+from pypdf import PdfWriter
+from utils import get_all_forecast_dirs
+
+
+def write_merged_pdf(
+    to_merge: list[Path | str], output_path: Path | str
+) -> None:
+    """
+    Merge an ordered list of PDF files and
+    write the result to a designated output path.
+
+    Parameters
+    ----------
+    to_merge
+        List of paths to the PDF files to merge.
+
+    output_path
+        Where to write the merged PDF.
+    """
+    pdf_writer = PdfWriter()
+    for pdf_file in to_merge:
+        pdf_writer.append(pdf_file)
+    pdf_writer.write(output_path)
+
+
+def merge_pdfs_from_subdirs(
+    base_dir: str | Path,
+    file_name: str,
+    output_file_name: str = None,
+    subdirs_only: list[str] = None,
+    subdir_pattern="*",
+) -> None:
+    """
+    Find matching PDF files from a set of
+    subdirectories of a base directory
+    and merge them, writing the resulting
+    merged PDF file the base directory.
+
+    Parameters
+    ----------
+    base_dir
+       The base directory in which to save
+       the resultant merged PDF file.
+
+    file_name
+       Name of the files to merge. Must be an
+       exact match.
+
+    output_file_name
+       Name for the merged PDF file, which will be
+       saved within ``base_dir``. If ``None``,
+       use ``file_name``. Default ``None``.
+
+    subdirs_only
+       Explicit list of subdirs to process. If
+       provided, process only subdirs found
+       within the ``base_dir`` that are named
+       in this list (and match the ``subdir_pattern``).
+       If ``None``, process all subdirs (provided
+       they match the ``subdir_pattern``).
+       Default ``None``.
+
+    subdir_pattern
+       Unix-shell style wildcard pattern that
+       subdirectories must match to be included.
+       Default ``'*'`` (match everything).
+       See documentation for :func:`fnmatch.fnmatch`
+       for details.
+
+    Returns
+    -------
+    None
+    """
+    subdirs = [
```

Comment:
Something like `[p.name for p in Path(base_dir).glob(subdir_pattern) if p.is_dir()]`

Reply:
Done in 1748b42